spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java

```diff
@@ -158,6 +162,11 @@ public class TaskTemplate implements TaskOperations {
 			this.thinExecutionsLink = null;
 		}
 
+		if(resources.getLink(THIN_EXECUTIONS_BY_NAME_RELATION).isPresent()) {
```

Comment:
This can be simplified w/ optional such as:
```java
this.thinExecutionsByNameLink = resources.getLink(THIN_EXECUTIONS_BY_NAME_RELATION).orElse(null);
```
This is fine as well though and its a backport.

Reply:
Done where appropriate